### PR TITLE
encodeURIComponent search component.

### PR DESCRIPTION
**Pull request by unjapones**
Linked to https://github.com/protofire/filecoin-CID-checker/issues/81

### Includes

- use `encodeURIComponent` in the UI the search term before before making the request to the API.
- Update Caddy setup: update `Caddyfile` with a minor change and set a specific version of the Caddy image in `docker-compose.yaml` so that the "slash" (Payload ID uri-encoded) path doesn't get decoded.
- Format (prettify) files in `ui` folder.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ services:
       - mongo
 
   caddy:
-    image: caddy
+    image: caddy:2.4.3-alpine
```
Comment (unjapones) [inline docker-compose.yaml:41]: We'll probably have to re-start things in the current online version to deploy this.

File: ui/src/hooks/useDeals.hook.ts [hunk 2]
```diff
@@ -10,27 +10,28 @@ export const useDeals = (
...
+  const crtReqHash = JSON.stringify([encodedSearch, query, activeFilter, verifiedFilter])
 
   useEffect(() => {
     let didCancel = false
 
     const run = async () => {
       try {
         if (reqHash !== crtReqHash) {
-          setDeals(RemoteData.loading());
+          setDeals(RemoteData.loading())
```
Comment (unjapones) [inline ui/src/hooks/useDeals.hook.ts:27]: This is the other significant change, the rest is prettify.

File: Caddyfile [hunk 3]
```diff
@@ -1,7 +1,9 @@
 :80 {
-  route /api/* {
-    uri strip_prefix api
+  handle_path /api/* {
     reverse_proxy http://cid-checker-api:3000
   }
   reverse_proxy http://cid-checker-frontend:5000
+  # log {
+  #  format single_field common_log
+  # }
```
Comment (unjapones) [inline Caddyfile:8]: These really don't affect the previous configuration: just found out that it was more succint and the commented `log` was useful during debugging/development.

**Comment by unjapones** [comment]
@smooth-operator pinging you because I'm not sure how's the process to deploy this fix in the current version of the CID-checker. I assumed that Digital-MOB's version is the most up-to-date one, so that's why I pushed the code to the repo I was able to contribute-to and then open the PR against the codebase that I think is running.

**Comment by ArseniiPetrovich** [comment]
Deployed this to the production server. I think we will need to integrate the fork changes into the upstream repo @unjapones

**Comment by unjapones** [comment]
Closing this PR. Check https://github.com/protofire/filecoin-CID-checker/issues/84 for more details.

**Pull request closed without merging**